src/components/fileSelect/FileSelect.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2021 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import React, { ChangeEvent, useRef, useState } from "react";
+
+import "styles/components/FileSelect.scss"
+import { Input } from "@dellstorage/clarity-react/forms/input/Input";
+import { Button } from "@dellstorage/clarity-react/forms/button";
+
+
+/**
+ * Props for the File select component
+ * @param {onFileChange} is the function triggered when a file is picked or changed.
+ * @param {accept} is a prop to get the file types which are to be accepted in the component.
+ * @param {multiple} is a prop which allows to select multiple files, if set to true
+ */
+export type FileSelectProps = {
+    onFileChange: (evt: ChangeEvent<HTMLInputElement>) => void,
+    accept?: string,
+    multiple?: boolean,
+}
+
+const FileSelect = (props: FileSelectProps) => {
+    const fileSelectRef = useRef<any>(null);
+
+    const [fileName, setFileName] = useState("")
```

Comment:
Can we convert this functional component into the class component as anyway you need to have some state here. ?